source/common/upstream/load_balancer_impl.h

```diff
@@ -236,6 +270,28 @@ class ZoneAwareLoadBalancerBase : public LoadBalancerBase {
 
   HostSet& localHostSet() const { return *local_priority_set_->hostSetsPerPriority()[0]; }
 
+  static HostsSource::SourceType localitySourceType(HostAvailability host_availability) {
+    switch (host_availability) {
+    case HostAvailability::Healthy:
+      return HostsSource::SourceType::LocalityHealthyHosts;
+    case HostAvailability::Degraded:
+      return HostsSource::SourceType::LocalityDegradedHosts;
+    default:
```

Comment:
nit: I don't think you need the default here and below. The compiler should know you covered all cases.

Reply:
That's what I thought too but it failed here: https://circleci.com/gh/envoyproxy/envoy/147261

My local compiler (osx) was okay with it, but seemed to fail in CI.

Reply:
Strange, no worries.